game/src/api/utils/DelayedSet.java

```diff
@@ -0,0 +1,94 @@
+package api.utils;
+
+import java.util.Collection;
+import java.util.ConcurrentModificationException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This class allows managing a Collection inside the System-Loops.
+ *
+ * <p>Since the Systems iterate over a Set of Entity's, a {@link ConcurrentModificationException} is
+ * normally thrown if this System adds/removes an entity of this set.
+ *
+ * <p>This class implements three different HashSets to allow delayed manipulation of the base Set.
+ *
+ * <p>{@link #current} contains the current "active" object of the collection. Use this Set to
+ * iterate over and work with. Use {@link #getSet()} to get this Set.
+ *
+ * <p>Use {@link #add} to add the given object to {@link #toAdd}. After the call of {@link #update},
+ * the objects inside this inner set will be added to {@link #current}
+ *
+ * <p>Use {@link #remove} to add the given object to {@link #toRemove}. After the call of {@link
+ * #update}, the objects inside this inner set will be removed from {@link #current}
+ *
+ * <p>Use {@link #update()} to add/remove the objects to/from {@link #current}. If you do this
+ * inside of an iteration over this Set, a {@link ConcurrentModificationException} will be thrown.
+ * It is best to use this right before or right after iterating over the set.
+ *
+ * @param <T> Type of the Elements to store in the Sets.
+ * @see ConcurrentModificationException
+ * @see HashSet
+ */
+public final class DelayedSet<T> {
+
+    private final Set<T> current = new HashSet<>();
+    private final Set<T> toAdd = new HashSet<>();
+    private final Set<T> toRemove = new HashSet<>();
+
+    /**
+     * Update the {@link #current} based on the elements in {@link #toAdd} and {@link #toRemove}.
+     *
+     * <p>Add all objects from {@link #toAdd} to {@link #current}. Remove all objects from {@link
+     * #toRemove} to {@link #current}. Clears {@link #toAdd} and {@link #toAdd}
+     */
+    public void update() {
+        current.addAll(toAdd);
+        current.removeAll(toRemove);
```

Comment:
Das ist eine spannende Design-Entscheidung. Wer hat Vorrang: Löschen oder Adden?

Reply:
Ich glaube es spielt eigentlich keine Rolle. 
Wenn die Dinger in `toAdd` oder `toRemove` hinterlegt sind, ist eh festgelegt, dass die Dinger hinzugefügt oder entfernt werden. Die Dinger existieren ja auch schon (halt irgendwo im Limbo). Verhalten können die sich auch erst im nächsten Frame.

Ich habs in der Javadoc aber mal explizit hervorgehoben.